uportal-war/src/main/java/org/jasig/portal/portlet/registry/PortletEntityIdStringUtils.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Licensed to Apereo under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work
+ * for additional information regarding copyright ownership.
+ * Apereo licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License.  You may obtain a
+ * copy of the License at the following location:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.jasig.portal.portlet.registry;
+
+import java.util.regex.Pattern;
+
+/**
+ * Utility class for defining methods to be used for dealing with portlet entity ID strings.
+ * 
+ * A portlet entity ID string consists of three parts separated by underscores.
+ * The format is:  <portlet-entity-id>_<layout-node-id>_<user-id>
+ * 
+ * Examples:
+ *   88_n149_52
+ *   71_u54_12
+ *  
+ * Note that if a portlet has a 'delegate' portlet, then the 'layout-node-id' will have the 
+ * format:  dlg-<delegate-portlet-window-id-with-underscores-replaced-by-dashes>
+ * 
+ * Examples:
+ *   90_dlg-5-ctf1-18_18      // here the portlet window ID of the delegate portlet is:  5_ctf1_18
+ *   75_dlg-10-ctf2-32.tw_32  // here the portlet entity ID of the delegate portlet is:  10_ctf2_32.tw
+ * 
+ * Currently, the admin interface uses Spring Webflow with 'delegate' portlets to display portlet configuration
+ * views.  When the admin interface portlet content is fetched with an 'exclusive' window state URL, the layout 
+ * node ID includes the "tw" transient (stateless) window instance ID.  This behavior was observed with RespondrJS 
+ * UI using 'exclusive' window state URLs.
+ * 
+ * @see flows/edit-portlet/configMode.jsp
+ * @see org.jasig.portal.portlet.delegation.*
+ * @see https://issues.jasig.org/browse/UP-2563
+ * @see PortletWindowIdStringUtils
+ * @see PortletWindowRegistryImpl#STATELESS_PORTLET_WINDOW_ID
+ * 
+ * @author Gary Roybal
+ * @version $Revision$
```

Comment:
I have been removing `@version $Revision$` from files that I touch because I have read that this metadata makes no sense in a git repo (where commit ids are not sequential).  Instead, I believe `@since` adds good value.